Dockerfile

```
@@ -265,8 +270,12 @@ RUN --mount=type=bind,from=flash-att-v2-cache,src=/usr/src/flash-attention-v2,ta
     pip install /usr/src/flash-attention-v2/*.whl --no-cache-dir
 
 # Copy over the auto-gptq wheel and install it
-RUN --mount=type=bind,from=auto-gptq-cache,src=/usr/src/auto-gptq-wheel,target=/usr/src/auto-gptq-wheel \
-    pip install /usr/src/auto-gptq-wheel/*.whl --no-cache-dir
+#RUN --mount=type=bind,from=auto-gptq-cache,src=/usr/src/auto-gptq-wheel,target=/usr/src/auto-gptq-wheel \
+#    pip install /usr/src/auto-gptq-wheel/*.whl --no-cache-dir
+
+# We only need to install a custom-built auto-gptq version if we need a pre-release
+# or are using a PyTorch nightly version
+RUN pip install auto-gptq=="${AUTO_GPTQ_VERSION}+cu118" --extra-index-url "https://huggingface.github.io/autogptq-index/whl/cu118/" --no-cache-dir
```

If we are updating to CUDA 12 at the same time
```suggestion
RUN pip install auto-gptq=="${AUTO_GPTQ_VERSION}" --no-cache-dir
```

---

@joerunde also it's possible we'll need to still build auto-gptq from source anyhow given the version differences of pytorch etc. See @cyang49's comment [here](https://github.com/IBM/text-generation-inference/pull/51#discussion_r1518460664).

---

Is it fine if that doesn't happen in this PR though?

---

Yes if it works then no problem!

---

The error I saw might be caused by the pre-built package doesn't contain Marlin build flag. Which may mean that eventually we'll have to build from source if Marlin support is desired.